lib/diagnostic.js

```diff
@@ -27,8 +27,12 @@ let access = null; // replace null with your response
 //
 //     `['alderaan', 'deathStar', 'fire', 'global']`
 //
-let scopes = [/* your response here */];
-//
+let scopes = ['deathStar'];
```

Comment:
`laserQuantity` is inside the `deathStar` scope, so only things that are also inside the `deathStar` scope can access `laserQuantity`. However, `fire` is defined inside the `deathStar` scope, so its function scope is inside the `deathStar` function scope - consequently, things inside the `fire` scope can access anything inside the `deathStar` scope.

Interestingly, that remains true even if we `return` the `fire` function and pass it out of `deathStar` -- that's the basis of closures.